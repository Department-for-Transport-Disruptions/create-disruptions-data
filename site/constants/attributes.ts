export const CD_SUMMARY = "summary";

export const CD_DESCRIPTION = "description";

export const CD_DISRUPTION_TYPE = "disruptionType";

export const CD_ASSOCIATED_LINK = "associatedLink";

export const CD_DISRUPTION_REASON = "disruptionReason";

export const CD_DISRUPTION_REAPEATS = "disruptionRepeats";

export const CD_NO_VALIDITY_END_DATE_TIME = "validityNoEndDateTime";

export const CD_NO_PUBLISH_END_DATE_TIME = "publishNoEndDateTime";

export const CD_VALIDITY_START_DATE_DAY = "validityStartDateDay";

export const CD_VALIDITY_START_DATE_MONTH = "validityStartDateMonth";

export const CD_VALIDITY_START_DATE_YEAR = "validityStartDateYear";

export const CD_VALIDITY_START_HOUR = "validityStartHour";

export const CD_VALIDITY_START_MINUTES = "validityStartMinute";

export const CD_VALIDITY_END_DATE_DAY = "validityEndDateDay";

export const CD_VALIDITY_END_DATE_MONTH = "validityEndDateMonth";

export const CD_VALIDITY_END_DATE_YEAR = "validityEndDateYear";

export const CD_VALIDITY_END_HOUR = "validityEndHour";

export const CD_VALIDITY_END_MINUTES = "validityEndMinute";

export const CD_PUBLISH_START_DATE_DAY = "publishStartDateDay";

export const CD_PUBLISH_START_DATE_MONTH = "publishStartDateMonth";

export const CD_PUBLISH_START_DATE_YEAR = "publishStartDateYear";

export const CD_PUBLISH_START_HOUR = "publishStartHour";

export const CD_PUBLISH_START_MINUTES = "publishStartMinute";

export const CD_PUBLISH_END_DATE_DAY = "publishEndDateDay";

export const CD_PUBLISH_END_DATE_MONTH = "publishEndDateMonth";

export const CD_PUBLISH_END_DATE_YEAR = "publishEndDateYear";

export const CD_PUBLISH_END_HOUR = "publishEndHour";

export const CD_PUBLISH_END_MINUTES = "publishEndMinute";
