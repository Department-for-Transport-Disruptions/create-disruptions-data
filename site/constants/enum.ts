export enum TransportMode {
    bus = "Bus",
    tram = "Tram",
    ferry = "Ferry",
    train = "Train",
}

export enum ConsequenceType {
    services = "Services",
    networkWide = "Network wide",
    operatorWide = "Operator wide",
    stops = "Stops",
}
