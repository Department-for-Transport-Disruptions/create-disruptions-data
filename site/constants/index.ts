import {
    MiscellaneousReason,
    PersonnelReason,
    EnvironmentReason,
    EquipmentReason,
} from "@create-disruptions-data/shared-ts/enums";

export const SUPPORT_EMAIL_ADDRESS = "bodshelpdesk@kpmg.co.uk";

export const SERVICE_EMAIL_ADDRESS = "bodshelpdesk@kpmg.co.uk";

export const SUPPORT_PHONE_NUMBER = "0800 123 1234";

export const DISRUPTION_REASONS: {
    value: MiscellaneousReason | PersonnelReason | EnvironmentReason | EquipmentReason;
    display: string;
}[] = [
    {
        value: MiscellaneousReason.roadWorks,
        display: "Road Works",
    },
    {
        value: MiscellaneousReason.vandalism,
        display: "Vandalism",
    },
    {
        value: MiscellaneousReason.routeDiversion,
        display: "Route Diversion",
    },
    {
        value: MiscellaneousReason.specialEvent,
        display: "Special Event",
    },
];

export const DISRUPTION_TYPES = ["planned", "unplanned"];

export const CD_DATE_FORMAT = "DD/MM/YYYY";
export const CONTACT_FEEDBACK_QUESTION = "Did you contact us for assistance at any point?";
export const SOLVE_FEEDBACK_QUESTION = "Did we solve your problem?";
export const HEAR_ABOUT_US_FEEDBACK_QUESTION = "How did you hear about our service?";
export const GENERAL_FEEDBACK_QUESTION = "Please let us know any feedback or suggestions for improvements you may have";

export const COOKIES_POLICY_COOKIE = "cdd-cookies-policy";

export const COOKIE_PREFERENCES_COOKIE = "cdd-cookie-preferences-set";

export const oneYearInSeconds = 31556952;

export const ID_TOKEN_COOKIE = "cdd-id-token";

export const COOKIES_DISRUPTION_INFO = "disruption-info";

export const COOKIES_DISRUPTION_ERRORS = "disruption-errors";

export const COOKIES_ADD_CONSEQUENCE_INFO = "cdd-consequence-info";

export const COOKIES_ADD_CONSEQUENCE_ERRORS = "cdd-consequence-errors";

export const CREATE_DISRUPTION_PAGE_PATH = "/create-disruption";

export const ERROR_PATH = "/_error";

export const ADD_CONSEQUENCE_PAGE_PATH = "/type-of-consequence";

export const STAGE = process.env.STAGE || "dev";
