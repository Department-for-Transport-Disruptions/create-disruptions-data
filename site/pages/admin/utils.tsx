import resendInvite from "../api/admin/resend-invite.api";

const createUserActionsLink = (
    key: string,
    index: number,
    username: string,
    userGroup: string,
    showResendInvite?: boolean,
) => {
    return (
        <>
            {showResendInvite ? (
                <>
                    <button
                        key={`${key}${index ? `-${index}` : ""}`}
                        className="govuk-link"
                        onClick={() => resendInvite(username, userGroup)}
                    >
                        Resend invite
                    </button>
                    <br />
                    <button
                        key={`${key}${index ? `-remove-${index}` : "-remove"}`}
                        className="govuk-link"
                        onClick={() => removeUser(username)}
                    >
                        Remove
                    </button>
                </>
            ) : (
                <button
                    key={`${key}${index ? `-${index}` : ""}`}
                    className="govuk-link"
                    onClick={() => removeUser(username)}
                >
                    Remove
                </button>
            )}
        </>
    );
};
