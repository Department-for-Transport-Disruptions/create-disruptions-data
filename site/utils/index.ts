import { ErrorInfo } from "interfaces";
import { ServerResponse } from "http";
import { NextApiResponse } from "next";

export const buildTitle = (errors: ErrorInfo[], title: string): string => {
    if (errors.length > 0) {
        return `Error: ${title}`;
    }

    return title;
};

export const redirectTo = (res: NextApiResponse | ServerResponse, location: string): void => {
    res.writeHead(302, {
        Location: location,
    });
    res.end();
};
