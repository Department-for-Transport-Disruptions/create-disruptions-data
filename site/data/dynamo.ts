import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
import { DynamoDBDocumentClient, QueryCommand } from "@aws-sdk/lib-dynamodb";
import { ptSituationElementSchema } from "@create-disruptions-data/shared-ts/siriTypes.zod";
import { notEmpty } from "../utils";
import logger from "../utils/logger";

const ddbDocClient = DynamoDBDocumentClient.from(new DynamoDBClient({ region: "eu-west-2" }));

const tableName = process.env.TABLE_NAME as string;

export const getDisruptionsDataFromDynamo = async () => {
    logger.info("Getting disruptions data from DynamoDB table...");

    const dbData = await ddbDocClient.send(
        new QueryCommand({
            TableName: tableName,
            KeyConditionExpression: "PK = :i",
            ExpressionAttributeValues: {
                ":i": "1",
            },
        }),
    );

    return dbData.Items?.map((item) => {
        const parsedItem = ptSituationElementSchema.safeParse(item);

        if (!parsedItem.success) {
            logger.error("Error parsing disruption from dynamo");
            logger.error(parsedItem.error.stack);
            return null;
        }

        return parsedItem.data;
    }).filter(notEmpty);
};
