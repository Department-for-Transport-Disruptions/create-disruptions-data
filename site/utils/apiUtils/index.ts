import { SocialMediaPostStatus } from "@create-disruptions-data/shared-ts/enums";
import cryptoRandomString from "crypto-random-string";
import { NextApiRequest, NextApiResponse } from "next";
import { parseCookies, setCookie } from "nookies";
import { z } from "zod";
import { IncomingMessage, ServerResponse } from "http";
import { notEmpty } from "..";
import {
    COOKIES_POLICY_COOKIE,
    COOKIE_CSRF,
    COOKIES_ID_TOKEN,
    COOKIE_PREFERENCES_COOKIE,
    DISRUPTION_DETAIL_PAGE_PATH,
    REVIEW_DISRUPTION_PAGE_PATH,
    COOKIES_REFRESH_TOKEN,
} from "../../constants";
import { publishToHootsuite } from "../../data/hootsuite";
import { sendTweet } from "../../data/twitter";
import { PageState } from "../../interfaces";
import { SocialMediaPost } from "../../schemas/social-media.schema";
import logger from "../logger";

export const setCookieOnResponseObject = (
    cookieName: string,
    cookieValue: string,
    res: NextApiResponse | ServerResponse<IncomingMessage>,
    lifetime?: number,
    httpOnly = true,
    sameSite: "lax" | "strict" | "none" = "lax",
): void => {
    // From docs: All cookies are httponly by default, and cookies sent over SSL are secure by
    // default. An error will be thrown if you try to send secure cookies over an insecure socket.
    setCookie({ res }, cookieName, cookieValue, {
        path: "/",
        sameSite,
        secure: process.env.NODE_ENV !== "development",
        maxAge: lifetime,
        httpOnly,
    });
};

export const destroyCookieOnResponseObject = (
    cookieName: string,
    res: NextApiResponse | ServerResponse<IncomingMessage>,
): void => {
    setCookieOnResponseObject(cookieName, "", res, 0);
};

export const redirectTo = (res: NextApiResponse | ServerResponse, location: string): void => {
    res.writeHead(302, {
        Location: location,
    });
    res.end();
};

export const redirectToError = (
    res: NextApiResponse | ServerResponse,
    message?: string,
    context?: string,
    error?: Error,
): void => {
    if (message && context && error) {
        logger.error(message, { context, error: error.stack });
    }

    redirectTo(res, "/500");
};

export const cleardownCookies = (req: NextApiRequest, res: NextApiResponse) => {
    const cookies = parseCookies({ req });

    const saveList = [
        COOKIES_POLICY_COOKIE,
        COOKIE_PREFERENCES_COOKIE,
        COOKIES_ID_TOKEN,
        COOKIES_REFRESH_TOKEN,
        COOKIE_CSRF,
    ];

    Object.keys(cookies).forEach((cookie) => {
        if (!saveList.includes(cookie)) {
            destroyCookieOnResponseObject(cookie, res);
        }
    });
};

export const getPageState = <T>(errorCookie: string, schemaObject: z.ZodType<T>, disruptionId?: string, data?: T) => {
    const inputsProps: PageState<Partial<T>> = {
        errors: [],
        inputs: {},
        disruptionId: disruptionId || "",
    };

    if (errorCookie) {
        return { ...(JSON.parse(errorCookie) as PageState<Partial<T>>), disruptionId: disruptionId || "" };
    }

    if (disruptionId) {
        const parsedData = schemaObject.safeParse(data);

        if (parsedData.success) {
            inputsProps.inputs = parsedData.data;
        }
    }

    return inputsProps;
};

export const getReturnPage = (req: NextApiRequest) => {
    const queryParam = req.headers.referer?.split("?")[1];
    const decodedQueryParam = queryParam ? decodeURIComponent(queryParam) : null;
    return decodedQueryParam?.includes(REVIEW_DISRUPTION_PAGE_PATH) ||
        decodedQueryParam?.includes(DISRUPTION_DETAIL_PAGE_PATH)
        ? queryParam
        : null;
};

export const isDisruptionFromTemplate = (req: NextApiRequest) => {
    const queryParam = req.headers.referer?.split("?")[1];
    const decodedQueryParam = queryParam ? decodeURIComponent(queryParam) : null;
    return decodedQueryParam?.includes("isFromTemplate") ? queryParam : null;
};

export const delay = (ms: number) => new Promise((res) => setTimeout(res, ms));

export const publishSocialMedia = async (
    socialMediaPosts: SocialMediaPost[],
    orgId: string,
    isUserStaff: boolean,
    canPublish: boolean,
) => {
    const socialMediaPromises = socialMediaPosts.map((post) => {
        if (post.status === SocialMediaPostStatus.pending) {
            if (post.accountType === "Twitter") {
                return sendTweet(orgId, post, isUserStaff, canPublish);
            }

            return publishToHootsuite(post, orgId, isUserStaff, canPublish);
        }

        return null;
    });

    await Promise.all(socialMediaPromises);
};

export const redirectToWithQueryParams = (
    req: NextApiRequest,
    res: NextApiResponse,
    queryParamsToForward: string[],
    location: string,
    paramsToAdd?: string[],
) => {
    const queryStringParams = queryParamsToForward
        .map((p) => {
            const paramValue = req.query[p];

            return paramValue ? `${p}=${paramValue.toString()}` : null;
        })
        .filter(notEmpty);

    redirectTo(
        res,
        `${location}${
            [...queryStringParams, ...(paramsToAdd || [])].length > 0
                ? `?${[...queryStringParams, ...(paramsToAdd || [])].join("&")}`
                : ""
        }`,
    );
};

export const formatCreateDisruptionBody = (body: object) => {
    const validity = Object.entries(body)
        .filter((item) => item.toString().startsWith("validity"))
        .map((arr: string[]) => {
            const [, values] = arr;

            return {
                disruptionStartDate: values[0],
                disruptionStartTime: values[1],
                disruptionEndDate: values[2],
                disruptionEndTime: values[3],
                disruptionNoEndDateTime: values[4],
                disruptionRepeats: values[5],
                disruptionRepeatsEndDate: values[6],
            };
        });

    const disruptionRepeatsEndDate = Object.entries(body)
        .filter((item) => item.toString().startsWith("disruptionRepeatsEndDate"))
        .map((arr: string[]) => {
            const [, values] = arr;
            let endDate = values;
            if (Array.isArray(values)) {
                // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
                endDate = values[0] ? values[0] : values[1];
            }
            return endDate;
        });

    const displayId: unknown[] = Object.entries(body)
        .filter((item) => item.includes("displayId"))
        .flat();

    const cleansedBody = Object.fromEntries(
        Object.entries(body).filter((item) => !item.toString().startsWith("validity")),
    );

    return {
        ...cleansedBody,
        validity,
        disruptionRepeatsEndDate: disruptionRepeatsEndDate ? disruptionRepeatsEndDate[0] : disruptionRepeatsEndDate,
        displayId: displayId && displayId.length > 1 && displayId[1] ? displayId[1] : cryptoRandomString({ length: 6 }),
    };
};
