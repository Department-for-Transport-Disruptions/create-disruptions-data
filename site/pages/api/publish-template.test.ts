import { PublishStatus } from "@create-disruptions-data/shared-ts/enums";
import MockDate from "mockdate";
import { describe, it, expect, afterEach, vi, afterAll, beforeEach } from "vitest";
import publishTemplate from "./publish-template.api";
import { ERROR_PATH, REVIEW_TEMPLATE_PAGE_PATH, VIEW_ALL_TEMPLATES_PAGE_PATH } from "../../constants/index";
import * as dynamo from "../../data/dynamo";
import { FullDisruption } from "../../schemas/disruption.schema";
import { Organisation, defaultModes } from "../../schemas/organisation.schema";
import {
    disruptionWithConsequencesAndSocialMediaPosts,
    getMockRequestAndResponse,
    disruptionWithNoConsequences,
    DEFAULT_ORG_ID,
    disruptionWithConsequences,
} from "../../testData/mockData";
import * as apiUtils from "../../utils/apiUtils";

const defaultDisruptionId = "acde070d-8c4c-4f0d-9d8a-162843c10333";

const orgInfo: Organisation = {
    name: "DepartmentForTransport",
    adminAreaCodes: ["001", "002"],
    mode: defaultModes,
};
describe("publish-template", () => {
    const writeHeadMock = vi.fn();
    vi.mock("../../utils/apiUtils", async () => ({
        ...(await vi.importActual<object>("../../utils/apiUtils")),
        setCookieOnResponseObject: vi.fn(),
        destroyCookieOnResponseObject: vi.fn(),
        cleardownCookies: vi.fn(),
        publishToHootsuite: vi.fn(),
    }));

    vi.mock("../../data/dynamo", () => ({
        insertPublishedDisruptionIntoDynamoAndUpdateDraft: vi.fn(),
        getDisruptionById: vi.fn(),
        getOrganisationInfoById: vi.fn(),
    }));

    vi.mock("crypto", () => ({
        randomUUID: () => "id",
    }));

    MockDate.set("2023-03-03");

    const insertDisruptionSpy = vi.spyOn(dynamo, "insertPublishedDisruptionIntoDynamoAndUpdateDraft");
    const getDisruptionSpy = vi.spyOn(dynamo, "getDisruptionById");
    const publishSocialMediaSpy = vi.spyOn(apiUtils, "publishSocialMedia");
    const getOrganisationInfoByIdSpy = vi.spyOn(dynamo, "getOrganisationInfoById");

    beforeEach(() => {
        getOrganisationInfoByIdSpy.mockResolvedValue(orgInfo);
    });

    afterEach(() => {
        vi.resetAllMocks();
    });

    afterAll(() => {
        MockDate.reset();
    });

    it("should retrieve valid data from cookies, write to dynamo and redirect", async () => {
        getDisruptionSpy.mockResolvedValue(disruptionWithConsequences);
        const { req, res } = getMockRequestAndResponse({
            body: {
                disruptionId: defaultDisruptionId,
            },
            mockWriteHeadFn: writeHeadMock,
        });

        await publishTemplate(req, res);

        expect(dynamo.insertPublishedDisruptionIntoDynamoAndUpdateDraft).toBeCalledTimes(1);
        expect(dynamo.insertPublishedDisruptionIntoDynamoAndUpdateDraft).toBeCalledWith(
            disruptionWithConsequences,
            DEFAULT_ORG_ID,
            PublishStatus.published,
            "test@example.com",
            undefined,
            true,
        );
        expect(writeHeadMock).toBeCalledWith(302, { Location: VIEW_ALL_TEMPLATES_PAGE_PATH });
    });

    it("should retrieve valid data from cookies, write to dynamo and redirect with social media", async () => {
        getDisruptionSpy.mockResolvedValue(disruptionWithConsequencesAndSocialMediaPosts);
        const { req, res } = getMockRequestAndResponse({
            body: {
                disruptionId: defaultDisruptionId,
            },
            mockWriteHeadFn: writeHeadMock,
        });

        publishSocialMediaSpy.mockResolvedValue();

        await publishTemplate(req, res);

        expect(dynamo.insertPublishedDisruptionIntoDynamoAndUpdateDraft).toBeCalledTimes(1);
        expect(dynamo.insertPublishedDisruptionIntoDynamoAndUpdateDraft).toBeCalledWith(
            disruptionWithConsequencesAndSocialMediaPosts,
            DEFAULT_ORG_ID,
            PublishStatus.published,
            "test@example.com",
            undefined,
            true,
        );
        expect(writeHeadMock).toBeCalledWith(302, { Location: VIEW_ALL_TEMPLATES_PAGE_PATH });
    });

    it("should redirect to error page if disruptionId not passed", async () => {
        getDisruptionSpy.mockResolvedValue(disruptionWithConsequencesAndSocialMediaPosts);
        const { req, res } = getMockRequestAndResponse({
            mockWriteHeadFn: writeHeadMock,
        });

        await publishTemplate(req, res);

        expect(dynamo.insertPublishedDisruptionIntoDynamoAndUpdateDraft).not.toBeCalled();
        expect(writeHeadMock).toBeCalledWith(302, { Location: ERROR_PATH });
    });

    it("should redirect to error page if disruption is invalid", async () => {
        getDisruptionSpy.mockResolvedValue({} as FullDisruption);
        const { req, res } = getMockRequestAndResponse({
            body: {
                disruptionId: defaultDisruptionId,
            },
            mockWriteHeadFn: writeHeadMock,
        });

        await publishTemplate(req, res);

        expect(dynamo.insertPublishedDisruptionIntoDynamoAndUpdateDraft).not.toBeCalled();
        expect(writeHeadMock).toBeCalledWith(302, { Location: ERROR_PATH });
    });

    it("should write the correct disruptions data to dynamoDB", async () => {
        getDisruptionSpy.mockResolvedValue(disruptionWithConsequencesAndSocialMediaPosts);
        const { req, res } = getMockRequestAndResponse({
            body: {
                disruptionId: disruptionWithConsequencesAndSocialMediaPosts.disruptionId,
            },
            mockWriteHeadFn: writeHeadMock,
        });

        await publishTemplate(req, res);

        expect(insertDisruptionSpy.mock.calls[0]).toMatchSnapshot();
    });

    it("should redirect to review page if no consequences", async () => {
        getDisruptionSpy.mockResolvedValue(disruptionWithNoConsequences);
        const { req, res } = getMockRequestAndResponse({
            body: {
                disruptionId: defaultDisruptionId,
            },
            mockWriteHeadFn: writeHeadMock,
        });

        await publishTemplate(req, res);

        expect(dynamo.insertPublishedDisruptionIntoDynamoAndUpdateDraft).not.toBeCalled();
        expect(writeHeadMock).toBeCalledWith(302, {
            Location: `${REVIEW_TEMPLATE_PAGE_PATH}/${defaultDisruptionId}`,
        });
    });
});
