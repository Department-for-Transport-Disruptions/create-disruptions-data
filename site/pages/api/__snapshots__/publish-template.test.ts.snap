// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`publish-template > should write the correct disruptions data to dynamoDB 1`] = `
[
  {
    "consequences": [
      {
        "consequenceIndex": 1,
        "consequenceType": "networkWide",
        "description": "Some consequence description",
        "disruptionId": "acde070d-8c4c-4f0d-9d8a-162843c10333",
        "disruptionSeverity": "slight",
        "removeFromJourneyPlanners": "no",
        "vehicleMode": "tram",
      },
      {
        "consequenceIndex": 0,
        "consequenceOperators": [
          {
            "operatorNoc": "FSYO",
            "operatorPublicName": "Operator Name",
          },
        ],
        "consequenceType": "operatorWide",
        "description": "Some consequence description",
        "disruptionDelay": "40",
        "disruptionId": "acde070d-8c4c-4f0d-9d8a-162843c10333",
        "disruptionSeverity": "severe",
        "removeFromJourneyPlanners": "yes",
        "vehicleMode": "bus",
      },
    ],
    "description": "Test description",
    "displayId": "8fg3ha",
    "disruptionId": "acde070d-8c4c-4f0d-9d8a-162843c10333",
    "disruptionNoEndDateTime": "true",
    "disruptionReason": "staffInWrongPlace",
    "disruptionStartDate": "18/03/2023",
    "disruptionStartTime": "1200",
    "disruptionType": "planned",
    "orgId": "35bae327-4af0-4bbf-8bfa-2c085f214483",
    "publishStartDate": "10/03/2023",
    "publishStartTime": "1200",
    "publishStatus": "DRAFT",
    "socialMediaPosts": [
      {
        "accountType": "Hootsuite",
        "disruptionId": "f8d602b9-6e09-4fd7-b14b-deb1ca5b4f24",
        "hootsuiteProfile": "127196025",
        "image": {
          "filepath": "/somefile/path",
          "key": "e9f6962b-1e77-4d0b-9cr2-f123315fd14c/r8e603b8-6e08-4fd7-b12b-deb1ca5b4g23/1.png",
          "mimetype": "image/png",
          "originalFilename": "test-image.png",
          "size": 70872,
        },
        "messageContent": "Test post 12345",
        "publishDate": "19/06/2023",
        "publishTime": "1805",
        "socialAccount": "13958638",
        "socialMediaPostIndex": 0,
        "status": "Successful",
      },
      {
        "accountType": "Hootsuite",
        "disruptionId": "f8d602b9-6e09-4fd7-b14b-deb1ca5b4f24",
        "hootsuiteProfile": "127196025",
        "image": {
          "filepath": "/somefile/path",
          "key": "e9f6962b-1e77-4d0b-9cr2-f123315fd14c/r8e603b8-6e08-4fd7-b12b-deb1ca5b4g23/1.png",
          "mimetype": "image/png",
          "originalFilename": "test-image.png",
          "size": 70872,
        },
        "messageContent": "Test post 1234567",
        "publishDate": "20/06/2023",
        "publishTime": "2005",
        "socialAccount": "137196026",
        "socialMediaPostIndex": 1,
        "status": "Pending",
      },
    ],
    "summary": "Some summary",
    "template": false,
    "validity": [
      {
        "disruptionEndDate": "17/03/2023",
        "disruptionEndTime": "1700",
        "disruptionStartDate": "10/03/2023",
        "disruptionStartTime": "1200",
      },
    ],
  },
  "35bae327-4af0-4bbf-8bfa-2c085f214483",
  "PUBLISHED",
  "test@example.com",
  undefined,
  true,
]
`;
